core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -192,7 +193,9 @@ class HadoopRDD[K, V](
     if (newInputFormat.isInstanceOf[Configurable]) {
       newInputFormat.asInstanceOf[Configurable].setConf(conf)
     }
-    HadoopRDD.putCachedMetadata(inputFormatCacheKey, newInputFormat)
+    if (shouldCacheInputFormat) {
+      HadoopRDD.putCachedMetadata(inputFormatCacheKey, newInputFormat)
```

Comment:
it seems that the current behavior of Spark is always invoking `HadoopRDD.putCachedMetadata(inputFormatCacheKey, newInputFormat)`

But since you set `spark.inputFormat.cache.enabled` to be `false` by default, this will change the current behavior of Spark, which brings some incompatibility to the current user